app/models/blacklist.rb

```diff
@@ -0,0 +1,28 @@
+class Blacklist
+  def initialize(possible_match)
+    @possible_match = possible_match
+  end
+
+  def blacklisted?
+    if possible_match.nil?
+      false
+    else
+      uri = URI.parse(possible_match)
+      host = normalize_host(uri)
+      BlacklistedReferrer.all.any? do |referrer|
+        prefix_uri = URI.parse(referrer.prefix)
+
+        host == normalize_host(prefix_uri) &&
+          uri.path.starts_with?(prefix_uri.path)
```

Comment:
I'm finding this a little confusing. I think the part that's getting me is that we're storing the prefixes as one thing, and then normalize i when we take it out with `normalize_host`. Could we:

* Store as the normalized host in the first place
* Use a query to test this instead of loading all BlacklistedReferrers into memory?

I'm not really worried about optimization, but I think it could read better. For example if we store all urls without their protocol (which I assume we control because we'll be entering them) then the `else` block can turn into:

```ruby
host = URI.parse(possible_match).host
BlacklistedReferrer.where("prefix LIKE ?", "#{host}%").
  or(BlacklistedReferrer.where("prefix LIKE ?", "www.#{host}%")).
  any?
```

Downside is it involves SQL but I think it gets the point across more succinctly without compromising much readability. I'd love to know what you think!

Reply:
Nice! I like this a lot better. I'm definitely doing this.